src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,199 @@
+        }
+
+        removeAllProductsFromOrder(order.getId());
+        insertProductsToOrder(order);
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders SET deleted = TRUE WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete order with id = "
+                    + id, ex);
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Long orderId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        return new Order(orderId, userId);
+    }
+
+    private List<Product> getListOfProducts(Long orderId) {
+        String query = "SELECT op.product_id AS id, p.product_name AS name, p.price "
+                + "FROM orders_products op "
+                + "INNER JOIN products p "
+                + "ON op.product_id = p.product_id "
+                + "WHERE op.order_id = ?;";
+        List<Product> products = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, orderId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long productId = resultSet.getLong("id");
+                String productName = resultSet.getString("name");
+                Long price = resultSet.getLong("price");
+                Product product = new Product(productId, productName, price);
+                products.add(product);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get products from order with id = "
+                    + orderId, ex);
+        }
+        return products;
+    }
+
+    private void removeAllProductsFromOrder(Long orderId) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, orderId);
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't remove products from order with id = "
+                    + orderId, ex);
+        }
+    }
+
+    private void insertProductsToOrder(Order order) {
+        String query = "INSERT INTO orders_products (order_id, product_id) VALUES (?, ?);";
+        for (Product product: order.getProducts()) {
+            try (Connection connection = ConnectionUtil.getConnection()) {
```

Comment:
move try before your loop, 8 point in checklist 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_23.html

Reply:
fixed